app/controllers/admin/facilities_controller.rb

```diff
@@ -9,23 +9,35 @@ class Admin::FacilitiesController < AdminController
     :validate_facility_rows, if: :file_exists?, only: [:upload]
 
   def index
-    authorize([:manage, :facility, Facility])
+    unless Flipper.enabled?(:user_roles)
+      authorize([:manage, :facility, Facility])
+    end
 
     if searching?
-      facilities = policy_scope([:manage, :facility, Facility]).search_by_name(search_query)
+      facilities = if Flipper.enabled?(:user_roles)
+        current_admin.facilities.search_by_name(search_query)
+      else
+        policy_scope([:manage, :facility, Facility]).search_by_name(search_query)
+      end
       facility_groups = FacilityGroup.where(facilities: facilities)
 
       @organizations = Organization.where(facility_groups: facility_groups)
       @facility_groups = facility_groups.group_by(&:organization)
       @facilities = facilities.group_by(&:facility_group)
     else
-      @organizations = policy_scope([:manage, :facility, Organization])
-      @facility_groups = @organizations.map { |organization|
-        [organization, policy_scope([:manage, :facility, organization.facility_groups])]
-      }.to_h
-      @facilities = @facility_groups.values.flatten.map { |facility_group|
-        [facility_group, policy_scope([:manage, :facility, facility_group.facilities])]
-      }.to_h
+      if Flipper.enabled?(:user_roles)
+        # @organizations = policy_scope([:manage, :facility, Organization])
```

Comment:
I don't think we would need to directly look up orgs here anymore, as I think it was just for the grouping part? Not sure.